### ISSUE TITLE
Include the volume name in the container name to prevent duplicates 

### ISSUE THREAD

**Issue by dducro** [tighten/takeout#77]
First of all, thanks for creating this easy tool!
This already was exactly my work flow, creating docker containers for just the services like ElasticSearch, Redis etc. This makes maintaining them a lot easier! :D

I'd like to create a separate Elastic search and Redis container for each project.
However, if they both use the same version there is a name conflict.

This could be solved by including the volume name in the container name like: `TO--redis--volume-name--buster`

What are your thoughts on this?
Thanks!

```
ERR  docker: Error response from daemon: Conflict. The container name "/TO--redis--buster" is already in use by container "0d0d690f64328b59d4a2b60f54cd99ee2254ba48f63db302b1738ca927347410". You have to remove (or rename) that container to be able to reuse that name.
See 'docker run --help'.
```

**Comment by mattstauffer** [comment]
I think this is a great idea! I'm trying to decide whether it's a breaking change or not. I.. sort of think it isn't? So.. we may be able to add this in a dot release.

**Comment by dducro** [comment]
Thanks for your answer! :)
I see that not all services have a volume, like: MsSql, Memcached, MailHog.
This means the container name can't always include the volume in its name.
In these cases it would still be nice to optionally be able to provide a name to make the containers unique.
I don't think it's a breaking change because all containers starting with `TO--` are listed an all commands are done with its ID.
Please let me know if you would like me to work on a PR.

(Just curious, doesn't MsSql need a volume as well?)

**Comment by josecanhelp** [comment]
@dducro MsSql could benefit from a volume, but the official image has trouble mounting docker volumes for some reason. In order to still have the service available in Takeout, we needed to disable volumes for that service.

I have seen some tricks to make volumes work with MsSql, but for now it's not available. So you can bring up the container, but as soon as you destroy it, the data will be removed.

We are looking into the option to bring down a container without destroying it though, so that will make working with MsSql a little better.

**Comment by mattstauffer** [comment]
The only way I can think of so far is to add an additional prompt to the setup that's basically the "nickname" or something, which would default to the volume name, and the nickname would always be appended to the base name (e.g. `TO--mysql--5.7--nicknamehere`)

**Comment by dducro** [comment]
That would be perfect! This way the user could have an instance per project if he so desires.
I also use the docker desktop app to list the containers and turn them on/off, to see the project name in the container name would be a great improvement! :)

**Comment by mattstauffer** [comment]
@dducro OK, it's a plan!

**Comment by nehero** [comment]
👋  Hey! I decided to take a stab at this in #89

**Comment by connor11528** [comment]
Thanks for opening this issue @dducro and the potential fix @ozziexsh. I'm looking to set up two Postgres databases, one for local development and one for running tests against. Currently getting this naming conflict error when trying to create my second Postgres db.

Workaround for me is to [rename](https://docs.docker.com/engine/reference/commandline/rename/) one of the containers. You can do this from the CLI:
```
$ docker rename TO--postgresql--13.0 my-database-name
```

**Comment by josecanhelp** [comment]
closed per reason in #89
